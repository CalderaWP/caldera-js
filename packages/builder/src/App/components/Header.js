import React from 'react';

const titleStyle = {
	alignContent: 'normal',
	alignItems: 'normal',
	alignSelf: 'auto',
	alignmentBaseline: 'auto',
	all: '',
	animation: 'none 0s ease 0s 1 normal none running',
	animationDelay: '0s',
	animationDirection: 'normal',
	animationDuration: '0s',
	animationFillMode: 'none',
	animationIterationCount: '1',
	animationName: 'none',
	animationPlayState: 'running',
	animationTimingFunction: 'ease',
	backfaceVisibility: 'visible',
	background:
		'rgba(0, 0, 0, 0) none repeat scroll 0% 0% / auto padding-box border-box',
	backgroundAttachment: 'scroll',
	backgroundBlendMode: 'normal',
	backgroundClip: 'border-box',
	backgroundColor: 'rgba(0, 0, 0, 0)',
	backgroundImage: 'none',
	backgroundOrigin: 'padding-box',
	backgroundPosition: '0% 0%',
	backgroundPositionX: '0%',
	backgroundPositionY: '0%',
	backgroundRepeat: 'repeat',
	backgroundRepeatX: '',
	backgroundRepeatY: '',
	backgroundSize: 'auto',
	baselineShift: '0px',
	blockSize: '103px',
	border: '',
	borderBlockEnd: '1px solid rgba(0, 0, 0, 0)',
	borderBlockEndColor: 'rgba(0, 0, 0, 0)',
	borderBlockEndStyle: 'solid',
	borderBlockEndWidth: '1px',
	borderBlockStart: '1px solid rgba(0, 0, 0, 0)',
	borderBlockStartColor: 'rgba(0, 0, 0, 0)',
	borderBlockStartStyle: 'solid',
	borderBlockStartWidth: '1px',
	borderBottom: '1px solid rgba(0, 0, 0, 0)',
	borderBottomColor: 'rgba(0, 0, 0, 0)',
	borderBottomLeftRadius: '0px',
	borderBottomRightRadius: '0px',
	borderBottomStyle: 'solid',
	borderBottomWidth: '1px',
	borderCollapse: 'separate',
	borderColor: 'rgba(0, 0, 0, 0)',
	borderImage: 'none',
	borderImageOutset: '0px',
	borderImageRepeat: 'stretch',
	borderImageSlice: '100%',
	borderImageSource: 'none',
	borderImageWidth: '1',
	borderInlineEnd: '1px solid rgba(0, 0, 0, 0)',
	borderInlineEndColor: 'rgba(0, 0, 0, 0)',
	borderInlineEndStyle: 'solid',
	borderInlineEndWidth: '1px',
	borderInlineStart: '0px solid rgba(0, 0, 0, 0)',
	borderInlineStartColor: 'rgba(0, 0, 0, 0)',
	borderInlineStartStyle: 'solid',
	borderInlineStartWidth: '0px',
	borderLeft: '0px solid rgba(0, 0, 0, 0)',
	borderLeftColor: 'rgba(0, 0, 0, 0)',
	borderLeftStyle: 'solid',
	borderLeftWidth: '0px',
	borderRadius: '0px',
	borderRight: '1px solid rgba(0, 0, 0, 0)',
	borderRightColor: 'rgba(0, 0, 0, 0)',
	borderRightStyle: 'solid',
	borderRightWidth: '1px',
	borderSpacing: '0px 0px',
	borderStyle: 'solid',
	borderTop: '1px solid rgba(0, 0, 0, 0)',
	borderTopColor: 'rgba(0, 0, 0, 0)',
	borderTopLeftRadius: '0px',
	borderTopRightRadius: '0px',
	borderTopStyle: 'solid',
	borderTopWidth: '1px',
	borderWidth: '1px 1px 1px 0px',
	bottom: 'auto',
	boxShadow: 'none',
	boxSizing: 'border-box',
	breakAfter: 'auto',
	breakBefore: 'auto',
	breakInside: 'auto',
	bufferedRendering: 'auto',
	captionSide: 'top',
	caretColor: 'rgb(25, 30, 35)',
	clear: 'none',
	clip: 'auto',
	clipPath: 'none',
	clipRule: 'nonzero',
	color: 'rgb(25, 30, 35)',
	colorInterpolation: 'srgb',
	colorInterpolationFilters: 'linearrgb',
	colorRendering: 'auto',
	columnCount: 'auto',
	columnFill: 'balance',
	columnGap: 'normal',
	columnRule: '0px none rgb(25, 30, 35)',
	columnRuleColor: 'rgb(25, 30, 35)',
	columnRuleStyle: 'none',
	columnRuleWidth: '0px',
	columnSpan: 'none',
	columnWidth: 'auto',
	columns: 'auto auto',
	contain: 'none',
	content: 'normal',
	counterIncrement: 'none',
	counterReset: 'none',
	cursor: 'text',
	cx: '0px',
	cy: '0px',
	d: 'none',
	direction: 'ltr',
	display: 'block',
	dominantBaseline: 'auto',
	emptyCells: 'show',
	fill: 'rgb(0, 0, 0)',
	fillOpacity: '1',
	fillRule: 'nonzero',
	filter: 'none',
	flex: '0 1 auto',
	flexBasis: 'auto',
	flexDirection: 'column',
	flexFlow: 'column nowrap',
	flexGrow: '0',
	flexShrink: '1',
	flexWrap: 'nowrap',
	float: 'none',
	floodColor: 'rgb(0, 0, 0)',
	floodOpacity: '1',
	font:
		'normal normal 700 normal 45px / 63px -apple-system, system-ui, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
	fontDisplay: '',
	fontFamily:
		'-apple-system, system-ui, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
	fontFeatureSettings: 'normal',
	fontKerning: 'auto',
	fontSize: '45px',
	fontStretch: '100%',
	fontStyle: 'normal',
	fontVariant: 'normal',
	fontVariantCaps: 'normal',
	fontVariantEastAsian: 'normal',
	fontVariantLigatures: 'normal',
	fontVariantNumeric: 'normal',
	fontVariationSettings: 'normal',
	fontWeight: '700',
	gap: 'normal normal',
	grid: 'none / none / none / row / auto / auto',
	gridArea: 'auto / auto / auto / auto',
	gridAutoColumns: 'auto',
	gridAutoFlow: 'row',
	gridAutoRows: 'auto',
	gridColumn: 'auto / auto',
	gridColumnEnd: 'auto',
	gridColumnGap: 'normal',
	gridColumnStart: 'auto',
	gridGap: 'normal normal',
	gridRow: 'auto / auto',
	gridRowEnd: 'auto',
	gridRowGap: 'normal',
	gridRowStart: 'auto',
	gridTemplate: 'none / none / none',
	gridTemplateAreas: 'none',
	gridTemplateColumns: 'none',
	gridTemplateRows: 'none',
	height: '103px',
	hyphens: 'manual',
	imageRendering: 'auto',
	inlineSize: '775.5px',
	isolation: 'auto',
	justifyContent: 'normal',
	justifyItems: 'normal',
	justifySelf: 'auto',
	left: 'auto',
	letterSpacing: 'normal',
	lightingColor: 'rgb(255, 255, 255)',
	lineBreak: 'auto',
	lineHeight: '63px',
	listStyle: 'disc outside none',
	listStyleImage: 'none',
	listStylePosition: 'outside',
	listStyleType: 'disc',
	margin: '0px',
	marginBlockEnd: '0px',
	marginBlockStart: '0px',
	marginBottom: '0px',
	marginInlineEnd: '0px',
	marginInlineStart: '0px',
	marginLeft: '0px',
	marginRight: '0px',
	marginTop: '0px',
	marker: '',
	markerEnd: 'none',
	markerMid: 'none',
	markerStart: 'none',
	mask: 'none',
	maskType: 'luminance',
	maxBlockSize: 'none',
	maxHeight: 'none',
	maxInlineSize: 'none',
	maxWidth: 'none',
	maxZoom: '',
	minBlockSize: '0px',
	minHeight: '0px',
	minInlineSize: '0px',
	minWidth: '0px',
	minZoom: '',
	mixBlendMode: 'normal',
	objectFit: 'fill',
	objectPosition: '50% 50%',
	offset: 'none 0px auto 0deg',
	offsetDistance: '0px',
	offsetPath: 'none',
	offsetRotate: 'auto 0deg',
	opacity: '1',
	order: '0',
	orientation: '',
	orphans: '2',
	outline: 'rgba(0, 0, 0, 0) solid 1px',
	outlineColor: 'rgba(0, 0, 0, 0)',
	outlineOffset: '0px',
	outlineStyle: 'solid',
	outlineWidth: '1px',
	overflow: 'hidden',
	overflowAnchor: 'auto',
	overflowWrap: 'break-word',
	overflowX: 'hidden',
	overflowY: 'hidden',
	overscrollBehavior: 'auto auto',
	overscrollBehaviorX: 'auto',
	overscrollBehaviorY: 'auto',
	padding: '19px 14px',
	paddingBlockEnd: '19px',
	paddingBlockStart: '19px',
	paddingBottom: '19px',
	paddingInlineEnd: '14px',
	paddingInlineStart: '14px',
	paddingLeft: '14px',
	paddingRight: '14px',
	paddingTop: '19px',
	page: '',
	pageBreakAfter: 'auto',
	pageBreakBefore: 'auto',
	pageBreakInside: 'auto',
	paintOrder: 'normal',
	perspective: 'none',
	perspectiveOrigin: '387.75px 51.5px',
	placeContent: 'normal normal',
	placeItems: 'normal normal',
	placeSelf: 'auto auto',
	pointerEvents: 'auto',
	position: 'static',
	quotes: '',
	r: '0px',
	resize: 'none',
	right: 'auto',
	rowGap: 'normal',
	rx: 'auto',
	ry: 'auto',
	scrollBehavior: 'auto',
	scrollMargin: '0px',
	scrollMarginBlock: '0px',
	scrollMarginBlockEnd: '0px',
	scrollMarginBlockStart: '0px',
	scrollMarginBottom: '0px',
	scrollMarginInline: '0px',
	scrollMarginInlineEnd: '0px',
	scrollMarginInlineStart: '0px',
	scrollMarginLeft: '0px',
	scrollMarginRight: '0px',
	scrollMarginTop: '0px',
	scrollPadding: 'auto',
	scrollPaddingBlock: 'auto',
	scrollPaddingBlockEnd: 'auto',
	scrollPaddingBlockStart: 'auto',
	scrollPaddingBottom: 'auto',
	scrollPaddingInline: 'auto',
	scrollPaddingInlineEnd: 'auto',
	scrollPaddingInlineStart: 'auto',
	scrollPaddingLeft: 'auto',
	scrollPaddingRight: 'auto',
	scrollPaddingTop: 'auto',
	scrollSnapAlign: 'none',
	scrollSnapStop: 'normal',
	scrollSnapType: 'none',
	shapeImageThreshold: '0',
	shapeMargin: '0px',
	shapeOutside: 'none',
	shapeRendering: 'auto',
	size: '',
	speak: 'normal',
	src: '',
	stopColor: 'rgb(0, 0, 0)',
	stopOpacity: '1',
	stroke: 'none',
	strokeDasharray: 'none',
	strokeDashoffset: '0px',
	strokeLinecap: 'butt',
	strokeLinejoin: 'miter',
	strokeMiterlimit: '4',
	strokeOpacity: '1',
	strokeWidth: '1px',
	tabSize: '8',
	tableLayout: 'auto',
	textAlign: 'start',
	textAlignLast: 'auto',
	textAnchor: 'start',
	textCombineUpright: 'none',
	textDecoration: 'none solid rgb(25, 30, 35)',
	textDecorationColor: 'rgb(25, 30, 35)',
	textDecorationLine: 'none',
	textDecorationSkipInk: 'auto',
	textDecorationStyle: 'solid',
	textIndent: '0px',
	textOrientation: 'mixed',
	textOverflow: 'clip',
	textRendering: 'auto',
	textShadow: 'none',
	textSizeAdjust: 'auto',
	textTransform: 'none',
	textUnderlinePosition: 'auto',
	top: 'auto',
	touchAction: 'auto',
	transform: 'none',
	transformBox: 'view-box',
	transformOrigin: '387.75px 51.5px',
	transformStyle: 'flat',
	transition: 'border 0.1s ease-out 0s, box-shadow 0.1s linear 0s',
	transitionDelay: '0s, 0s',
	transitionDuration: '0.1s, 0.1s',
	transitionProperty: 'border, box-shadow',
	transitionTimingFunction: 'ease-out, linear',
	unicodeBidi: 'normal',
	unicodeRange: '',
	userSelect: 'auto',
	userZoom: '',
	vectorEffect: 'none',
	verticalAlign: 'baseline',
	visibility: 'visible',
	whiteSpace: 'pre-wrap',
	widows: '2',
	width: '775.5px',
	willChange: 'auto',
	wordBreak: 'keep-all',
	wordSpacing: '0px',
	wordWrap: 'break-word',
	writingMode: 'horizontal-tb',
	x: '0px',
	y: '0px',
	zIndex: 'auto',
	zoom: '1',
};

export const Header = ({ title, children }) => {
	return (
		<header>
			<h1 style={titleStyle}>{title}</h1>
			{children}
		</header>
	);
};
Header.defaultProps = {
	title: 'Caldera Forms',
};
