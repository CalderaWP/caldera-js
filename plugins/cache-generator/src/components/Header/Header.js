import React, {Component} from 'react';
import logo from "../../logo.svg";
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * The application header
 * @param title
 * @param className
 * @return {*}
 * @constructor
 */
export const Header = ({
						   title,
						   className
					   }) => (
	<header
		className={classNames("App-header", className)}>
		<img src={logo} className="App-logo" alt="logo"/>
		<h1 className="App-title">{title}</h1>
	</header>
);

/**
 * Prop definitions for Header component
 *
 * @type {{title: shim}}
 */
Header.propTypes = {
	title: PropTypes.string,
};

/**
 * Default prop types for header component
 *
 * @type {{title: string}}
 */
Header.defaultProps = {
	title: 'Welcome to Caldera WordPress Plugin'
};


