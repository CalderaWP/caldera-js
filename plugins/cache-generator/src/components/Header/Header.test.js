import React, {Fragment} from 'react';
import renderer from 'react-test-renderer';
import {Header} from "./Header";

describe( 'Header component', () => {
	it( 'Matches snapshot with title prop', () => {
		const component = renderer.create(
			<Header title={'Welcome To An Exciting Web Application'} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'Uses default title', () => {
		const component = renderer.create(
			<Header />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
