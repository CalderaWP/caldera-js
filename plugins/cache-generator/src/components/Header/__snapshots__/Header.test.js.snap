// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Matches snapshot 1`] = `
<header
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <h1
    className="App-title"
  >
    Welcome To An Exciting Web Application
  </h1>
</header>
`;

exports[`Header component Matches snapshot with title prop 1`] = `
<header
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <h1
    className="App-title"
  >
    Welcome To An Exciting Web Application
  </h1>
</header>
`;

exports[`Header component Uses default title 1`] = `
<header
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <h1
    className="App-title"
  >
    Welcome to Caldera WordPress Plugin
  </h1>
</header>
`;
